pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -712,8 +713,24 @@ func applyLoadBalancer(c *cluster.Cluster, lb *networking.LoadBalancerSettings,
 	case networking.LoadBalancerSettings_ROUND_ROBIN:
 		applyRoundRobinLoadBalancer(c, lb)
 	case networking.LoadBalancerSettings_PASSTHROUGH:
-		c.LbPolicy = cluster.Cluster_CLUSTER_PROVIDED
-		c.ClusterDiscoveryType = &cluster.Cluster_Type{Type: cluster.Cluster_ORIGINAL_DST}
+		// This can happen two cases.
+		// 1. A ServiceEntry is created with Resolution as DNS/STATIC and the DestinationRule
+		//    is configured with PASSTHROUGH load balancer.
+		// 2. A regular Kubernetes Service (of type EDS), configures DestinationRule with PASSTHROUGH load balancer.
```

Comment:
Not sure i fully understand, then a regular k8s service plus a pass through DR setting will be no redirect at all? That seems a misconfig too.

Reply:
As I mentioned there are use cases for regular K8S service to configure PASSTHROUGH if you want direct pod -> pod connectivity.  There are people that use this (we are one among them :-)) https://discuss.istio.io/t/503-between-pod-to-pod-communication-1-5-1/6121/15 is another example

Reply:
pod to pod communicate in k8s usually make use of headless service

Reply:
FYI https://kubernetes.io/docs/concepts/services-networking/dns-pod-service/#pod-s-hostname-and-subdomain-fields

Reply:
I know. This is to avoid that headless service and as I said people already use it because PASSTHROUGH is supported in DR.

Reply:
I donot think so, this is influcing normal k8s service. for k8s service, the cluster type is original_dstm why not check it here?

Reply:
This is currently supported for regular k8s service. As I said people use it already. We can not ask, them to move all to headless now - if people have both use cases, they have to create two services one with headless and and one with cluster ip. This is a breaking change. This change is to support that case.

For service entries, it is not supported today also - as cluster will be NACKed.

Reply:
Can you tell how people connect to one specific pod?

Reply:
They get IP of pod by some means (it is application dependent) and they will call using IP:PORT. See the second appraoch documented here https://discuss.istio.io/t/503-between-pod-to-pod-communication-1-5-1/6121/15 - create a subset and with PASSTHROUGH and call with ip:port

Reply:
Confused if directly access by ip, how does it select a cluster?